### PR TITLE
Fix double free in proc_pid module

### LINKED ISSUE #2: Occaisional SEG faults
Sometimes the tests fail due to a segfault, running locally this can be as infrequent as 1 in 10 to 20 runs.

So, no idea what's causing this - it makes me suspicious of the stability/usability of the underlying libproc.a library on more recent Mac OS X versions. But the 'C' version never seems to fail, so I'm not sure....

Needs investigation that's for sure.

### PR DESCRIPTION
**Pull request by lfairy**
The original code called `.as_ptr()` (which *borrows* its argument) and passed it to `Vec::from_raw_parts()` (which *consumes* it). So the underlying buffer would have ended up with two owners, which is wrong.

This patch changes the code to use `.set_len()` instead, which borrows the allocation instead of consuming it. It also uses `.as_mut_ptr()` instead of `.as_ptr()` for const-correctness.

Closes #2

### CODE REVIEW

**Comment by lfairy** [comment]
By the way, can you please check that this works on your machine? I don't have access to a macOS machine right now, so I only have Travis to go on.

File: src/libproc/proc_pid.rs [hunk 1]
```diff
@@ -214,10 +214,10 @@ pub fn listpids(proc_types: ProcType) -> Result<Vec<u32>, String> {
         Err(get_errno_with_message(ret))
     } else {
         unsafe {
-            pids = Vec::from_raw_parts(buffer_ptr as *mut u32, ret as usize, pids.capacity());
-            // Seems that the buffer is padded with a lot of pids set to zero
-            pids.retain(|&p| p > 0u32);
+            pids.set_len(ret as usize);
         }
+        // Seems that the buffer is padded with a lot of pids set to zero
+        pids.retain(|&p| p > 0);
```
Comment (lfairy) [inline src/libproc/proc_pid.rs]: I'm not sure if this `.retain()` call is necessary now -- the extra zeros may have been due to memory unsafety bugs before.

**Comment by andrewdavidmackenzie** [comment]
I've now got a few warnings, but that's because I've moved to rust 1.14.0 ..... so will try and fix them in a new PR.

Thanks!

File: src/libproc/proc_pid.rs (hunk 1 above)
Comment (andrewdavidmackenzie) [inline src/libproc/proc_pid.rs]: Let's take a look at that separately...?

**Comment by lfairy** [comment]
Cool beans! Feel free to ping me if you need any help.

**Comment by andrewdavidmackenzie** [comment]
BTW: would you want/like/mind being included in a list of contributors to the project?

**Comment by lambda-fairy** [comment]
Sure, I would appreciate it!

**Pull request merged**